nucleus/common/glassfish-api/src/main/java/org/glassfish/api/admin/AdminCommandLock.java

```diff
@@ -590,27 +590,23 @@ private void semaphoreWait(Semaphore s, String logMsg) {
          * @param r A Runnable which will be invoked by the method after the lock is suspended
          */
         public static void runWithSuspendedLock(Runnable r) {
-           
-            Lock lock = null;
-
+            // We need to determine the type of lock this thread holds.
+            Lock lock = rwlock.isWriteLockedByCurrentThread()
+                ? rwlock.writeLock() 
+                : rwlock.getReadHoldCount() > 0 ? rwlock.readLock() : null;
+            if (lock == null) {
+                // Not locked, run plain:
+                r.run();
+                return;
+            }
             try {
-                // We need to determine the type of lock this thread holds.
-                if (rwlock.isWriteLockedByCurrentThread()) {
-                    lock = rwlock.writeLock();
-                } else if (rwlock.getReadHoldCount() > 0) {
-                    lock = rwlock.readLock();
-                }
-
-                if (lock != null)
-                    lock.unlock();
-
+                lock.unlock();
                 // Run the caller's commands without a lock in place.
                 r.run();
             } finally {
                 // Relock before returning.  This may block if someone else
                 // already grabbed the lock.
-                if (lock != null)
-                    lock.lock();
```

Comment:
This is best understood looking at the method name. The method should execute the `Runnable` in a context where the lock is not locked. So if no lock is active we can run `r` plain, otherwise we unlock, run and lock again (in a way that static analysis can follow).